src/mesos_state_client.erl

```diff
@@ -36,6 +38,54 @@ maybe_enable_ssl(Options) ->
       Options
   end.
 
+-spec(maybe_use_token(list()) -> list()).
+maybe_use_token(Headers) ->
```

Comment:
Also, I’d refactor your flow a little bit
I would make
```maybe_use_token(Headers) ->
 +  case try_get_token() of
 +    undefined ->
 +      Headers;
 +    {ok, Token} ->
 +      [{"Authorization", Token}|Headers]
 +  end.
```
and then try_get_token can try getting it from an env variable, or a file